internal/scalibrextract/filesystem/vendored/vendored.go

```diff
@@ -0,0 +1,179 @@
+package vendored
+
+import (
+	"context"
+	//nolint:gosec
+	// md5 used to identify files, not for security purposes
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	scalibrfs "github.com/google/osv-scalibr/fs"
+	"github.com/google/osv-scalibr/plugin"
+	"github.com/google/osv-scalibr/purl"
+	"github.com/google/osv-scanner/pkg/osv"
+)
+
+var (
+	vendoredLibNames = map[string]struct{}{
+		"3rdparty":    {},
+		"dep":         {},
+		"deps":        {},
+		"thirdparty":  {},
+		"third-party": {},
+		"third_party": {},
+		"libs":        {},
+		"external":    {},
+		"externals":   {},
+		"vendor":      {},
+		"vendored":    {},
+	}
+)
+
+const (
+	// This value may need to be tweaked, or be provided as a configurable flag.
+	determineVersionThreshold = 0.15
+	maxDetermineVersionFiles  = 10000
+)
+
+type Extractor struct {
+	// ScanGitDir determines whether a vendored library with a git directory is scanned or not,
+	// this is used to avoid duplicate results, once from git scanning, once from vendoredDir scanning
+	ScanGitDir bool
+	// TODO(v2): Client rework
+	// determineVersionsClient
+}
+
+var _ filesystem.Extractor = Extractor{}
+
+// Name of the extractor.
+func (e Extractor) Name() string { return "filesystem/vendored" }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities {
+	return &plugin.Capabilities{}
+}
+
+// FileRequired returns true for .package-lock.json files under node_modules
+func (e Extractor) FileRequired(fapi filesystem.FileAPI) bool {
+	// Check if parent directory is one of the vendoredLibName
+	// Clean first before Dir call to avoid trailing slashes causing problems
+	parentDir := filepath.Base(filepath.Dir(filepath.Clean(fapi.Path())))
+	_, ok := vendoredLibNames[parentDir]
+	if !ok {
+		return false
+	}
+
+	// Stat costs performance, so perform it after the name check
+	stat, err := fapi.Stat()
+	if err != nil {
+		return false
+	}
+
+	return stat.IsDir()
+}
+
+// Extract extracts packages from yarn.lock files passed through the scan input.
```

Comment:
nit: update comment.

Reply:
Done, also updated some other comments in the other extractors as they were outdated as well.